### PR TITLE
Update .codecov.yml

### PR DESCRIPTION
**Pull request by stevepeak**
Fixes mapping coverage reports back to src not test

### CODE REVIEW

**Comment by wilzbach** [comment]
> Fixes mapping coverage reports back to src not test

Thanks at having a look at this @stevepeak - highly appreciated :)
Could you please squash your commits into one?
We use a merge bot and thus don't support squashed commits (apart from that [squashed commit by Github screw up the history](http://forum.dlang.org/post/mciqgandxypjwblexqaf@forum.dlang.org]))

File: .codecov.yml [hunk 1]
```diff
@@ -5,3 +5,8 @@ codecov:
   # DAutoTester and auto-tester are not in the default list
   ci:
     - !travis
+
+coverage:
+  fixes:
+    - "test/.*/::src/"
+    - "fail_compilation/::src/"
```
Comment (wilzbach) [inline .codecov.yml]: This shouldn't be necessary - in fact everything in `extra_files` can be safely ignored as they are stored `.lst` coverage files which are used to test the coverage counting.

Comment (stevepeak) [inline .codecov.yml]:
```
src/file.d
test/blah/file.d

# path=file.lst
.....
```

The issue is that codecov will resolve the path `file.lst` to `test/blah/file.d`.

Comment (wilzbach) [inline .codecov.yml]: I just referred to the second line, but I guess no one minds if it's necessary for your side ;-)

**Comment by ibuclaw** [comment]
Seems ok to me. I wouldn't worry about having multiple commits in a PR.

**Comment by wilzbach** [comment]
> Seems ok to me. I wouldn't worry about having multiple commits in a PR.

Oh I thought that's the policy to prevent a huge commit changelog. 
Well I guess the people reading the commit messages are quite entertained anyhows, for example with 
https://github.com/dlang/phobos/pull/4662

**Comment by ibuclaw** [comment]
You should grep for dwarf in dmds history.  ;-)

**Pull request merged**